File: dominoes-web-app/src/lib/shaper-domino/measurementHelpers.ts
import { Measurement } from './measurement';
import { UnitOfMeasure } from './unitOfMeasure';

export function inches(value: number): Measurement {
	return new Measurement(value, UnitOfMeasure.Inches);
}

export function millimeters(value: number): Measurement {
	return new Measurement(value, UnitOfMeasure.Millimeters);
}

export function centimeters(value: number): Measurement {
	return new Measurement(value * 10, UnitOfMeasure.Millimeters);
}

File: dotnet/DominoFactory/MeasurementHelpers.cs
// <copyright file="MeasurementHelpers.cs">Copyright (c) Peter Rosser. All rights reserved.</copyright>

namespace ShaperDominoLib;

public static class MeasurementHelpers
{
    public static Measurement Inches(this double value) => new(value, UnitOfMeasure.Inches);
    public static Measurement Millimeters(this double value) => new(value, UnitOfMeasure.Millimeters);
    public static Measurement Centimeters(this double value) => new(value * 10, UnitOfMeasure.Millimeters);

    public static Measurement Inches(this int value) => new(value, UnitOfMeasure.Inches);
    public static Measurement Millimeters(this int value) => new(value, UnitOfMeasure.Millimeters);
    public static Measurement Centimeters(this int value) => new(value * 10, UnitOfMeasure.Millimeters);
}
